crates/hyperswitch_connectors/src/connectors/nexixpay/transformers.rs

```diff
@@ -1,80 +1,405 @@
+#[serde(rename_all = "camelCase")]
+pub struct UpdateNexixpayConnectorMetaData {
+    pub three_d_s_auth_result: Option<ThreeDSAuthResult>,
+    pub three_d_s_auth_response: Option<String>,
+    pub authorization_operation_id: Option<String>,
+    pub capture_operation_id: Option<String>,
+    pub cancel_operation_id: Option<String>,
+    pub psync_flow: Option<NexixpayPaymentIntent>,
+    pub meta_data: serde_json::Value,
+    pub is_auto_capture: bool,
+}
+
+fn update_nexi_meta_data(
+    update_request: UpdateNexixpayConnectorMetaData,
+) -> CustomResult<serde_json::Value, errors::ConnectorError> {
+    let nexixpay_meta_data =
+        serde_json::from_value::<NexixpayConnectorMetaData>(update_request.meta_data)
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+
+    Ok(serde_json::json!(NexixpayConnectorMetaData {
+        three_d_s_auth_result: update_request
+            .three_d_s_auth_result
+            .or(nexixpay_meta_data.three_d_s_auth_result),
+        three_d_s_auth_response: update_request
+            .three_d_s_auth_response
+            .or(nexixpay_meta_data.three_d_s_auth_response),
+        authorization_operation_id: update_request
+            .authorization_operation_id
+            .clone()
+            .or(nexixpay_meta_data.authorization_operation_id.clone()),
+        capture_operation_id: {
+            if update_request.is_auto_capture {
+                update_request
+                    .authorization_operation_id
+                    .clone()
+                    .or(nexixpay_meta_data.authorization_operation_id)
+            } else {
+                update_request.capture_operation_id
+            }
+            .or(nexixpay_meta_data.capture_operation_id)
+        },
+        cancel_operation_id: update_request
+            .cancel_operation_id
+            .or(nexixpay_meta_data.cancel_operation_id),
+        psync_flow: update_request
+            .psync_flow
+            .unwrap_or(nexixpay_meta_data.psync_flow)
+    }))
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ThreeDSAuthData {
+    three_d_s_auth_response: String,
+    authentication_value: Secret<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPreProcessingResponse {
+    operation: Operation,
+    three_d_s_auth_result: ThreeDSAuthResult,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Operation {
+    additional_data: AdditionalData,
+    customer_info: CustomerInfo,
+    operation_amount: String,
+    operation_currency: Currency,
+    operation_id: String,
+    operation_result: NexixpayPaymentStatus,
+    operation_time: String,
+    operation_type: NexixpayOperationType,
+    order_id: String,
+    payment_circuit: String,
+    payment_end_to_end_id: String,
+    payment_instrument_info: String,
```

Comment:
Are these sensitive, shall these be wrapped under a `Secret`

Reply:
removed these fields as they were not getting used anywhere